ExecuteJobs.java

```diff
@@ -0,0 +1,80 @@
+import java.util.*;
+import java.io.*;
+
+public class ExecuteJobs {
+  public long start(String process) throws IOException {
+    // TODO some error handling
+    Process p = Runtime.getRuntime().exec(process);
+    long pid = p.pid();
+    return pid;
+  }
+
+  public int stop(long pid) throws IOException {
+    Process p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
+    BufferedReader pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+    String s = pInput.readLine();
+    if(s == null) {
+      // pid not found
+      return 2;
+    }
+
+    Runtime.getRuntime().exec("kill " + pid);
+    p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
+    pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+    s = pInput.readLine();
+    if(s != null) {
+      for(int i = 0; i < 5; i++) {
+        s = null;
+        p = Runtime.getRuntime().exec("kill " + pid);
+        pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+        s = pInput.readLine();
+        if(s == null) {
+          break;
+        }
+      }
+      s = null;
+      pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
+      s = pInput.readLine();
+      if(s != null) {
+        // Failed to kill pid
+        return 1;
+      }
+    }
+    // Successfully killed pid
+    return 0;     
+  }
+
+  public String query(long pid) throws IOException {
+    Process p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
```

Comment:
same here, seems like this logic (ps -q) and parsing of the subsequent output could be factored out in the separate function